client/src/components/AddVideo.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+
+const AddVideo = ({ onAddVideo }) => {
+  const [title, setTitle] = useState('');
+  const [url, setUrl] = useState('');
+
+  const getYouTubeVideoIdFromUrl = (url) => {
+    // Define patterns for different YouTube URL formats
+    const patterns = [
+      /(?:https?:\/\/)?(?:www\.)?youtube\.com\/watch\?v=([a-zA-Z0-9_-]+)/,
+      /(?:https?:\/\/)?(?:www\.)?youtu\.be\/([a-zA-Z0-9_-]+)/,
+    ];
+
+    // Iterate through patterns and try to match with the given URL
+    for (const pattern of patterns) {
+      const match = url.match(pattern);
+      if (match && match[1]) {
+        return match[1]; // Return the video ID
+      }
+    }
+
+    // If no match is found, return null
+    return null;
+  };
+
+  const handleAddVideo = () => {
+    if (title.trim() === '' || !isValidYouTubeUrl(url)) {
+      alert('Please enter a valid title and a valid YouTube URL.');
+      return;
+    }
+
+    // Validate if the URL is a YouTube video URL and extract the video ID
+    const videoId = getYouTubeVideoIdFromUrl(url);
+
+    // Generate a unique ID using the current timestamp
+    const newVideo = {
+      id: Date.now(),
+      title,
+      youtubeVideoId: videoId,
+      votes: 0,
+      uploadDate: new Date(),
+    };
+
+    // Call the parent component's function to add the video
+    onAddVideo(newVideo);
+
+    // Reset the input fields
+    setTitle('');
+    setUrl('');
+  };
+
+  // Function to check if a string is a valid YouTube URL
+  const isValidYouTubeUrl = (url) => {
```

Comment:
It feels like your `getYouTubeVideoIdFromUrl` function is doing better validation than this function - is this function really worth having/calling?